File: program48.c
//problem to count number of digits
#include<stdio.h>

int CountDigits(int iNo)
{
    int iCnt=0,iDigit=0;
    if(iNo<0)
    {
        iNo=-iNo;
    }
    while(iNo>0)        //this logic is same in all digit problems
    {
       iDigit=iNo%10;
       iCnt++;
       iNo=iNo/10;
    }  

    return iCnt;

}

int main()
{
    int iValue=0,iRet=0;

    printf("Enter number\n");
    scanf("%d",&iValue);

    iRet=CountDigits(iValue);
    printf("Number of digits are %d\n",iRet);

    return 0;
}

File: program48.java
//program to count number of digits
import java.lang.*;
import java.util.*;


class Count
{
    public int CountDigits(int no)
    {
        int i=0,cnt=0,digit=0;

        if(no<0)
        {
            no=-no;
        }

        while(no>0)
        {
            cnt++;
            no=no/10;
        }
        return cnt;
    }
}


class program48
{
    public static void main(String arg[])
    {
        Count obj=new Count();
        int no=0,ret=0;
        Scanner sobj=new Scanner(System.in);
        System.out.println("Enter number");
        no=sobj.nextInt();
        ret=obj.CountDigits(no);
        System.out.println("Number of digits are :"+ret);

       
        

    }
}